crates/meroctl/src/cli/context/call.rs

```diff
@@ -0,0 +1,185 @@
+use calimero_primitives::alias::Alias;
+use calimero_primitives::context::ContextId;
+use calimero_primitives::identity::PublicKey;
+use calimero_server_primitives::admin::CallContextResponse;
+use clap::Parser;
+use eyre::{bail, Result as EyreResult};
+use libp2p::Multiaddr;
+use regex::Regex;
+use reqwest::Client;
+
+use crate::cli::Environment;
+use crate::common::{do_request, fetch_multiaddr, load_config, multiaddr_to_url, RequestType, lookup_alias};
+use crate::output::{ErrorLine, InfoLine, Report};
+
+#[derive(Debug, Parser)]
+#[command(about = "Call a context function")]
+pub struct CallCommand {
+    #[clap(help = "The context ID or alias to call")]
+    context: String,
+
+    #[clap(help = "The function to call")]
+    function: String,
+
+    #[clap(help = "The parameters to pass to the function. Supports alias substitution with %alias% syntax")]
+    params: Option<String>,
+
+    #[clap(long = "as", help = "The identity alias to use for the call")]
+    identity: Option<Alias<PublicKey>>,
+}
+
+impl Report for CallContextResponse {
+    fn report(&self) {
+        if let Some(result) = &self.data.result {
+            println!("Result: {}", String::from_utf8_lossy(result));
+        } else {
+            println!("No result returned");
+        }
+    }
+}
+
+impl CallCommand {
+    pub async fn run(self, environment: &Environment) -> EyreResult<()> {
+        let config = load_config(&environment.args.home, &environment.args.node_name)?;
+        let multiaddr = fetch_multiaddr(&config)?;
+        let client = Client::new();
+
+        // Resolve context ID if it's an alias
+        let context_id = if let Ok(context_id) = self.context.parse::<ContextId>() {
+            context_id
+        } else if let Ok(alias) = self.context.parse::<Alias<ContextId>>() {
+            match lookup_alias(multiaddr.clone(), &config.identity, alias, None).await {
+                Ok(response) => {
+                    if let Some(context_id) = response.data.value {
+                        context_id
+                    } else {
+                        bail!("Context alias '{}' not found", self.context);
+                    }
+                }
+                Err(e) => bail!("Error looking up context alias '{}': {}", self.context, e),
+            }
+        } else {
+            bail!("Invalid context ID or alias format: {}", self.context);
+        };
```

Comment:
There are examples in our repo how to do this more cleanly, for example:
https://github.com/calimero-network/core/blob/master/crates/meroctl/src/cli/context/get.rs#L82-#L86 
```suggestion
       let context_id = resolve_alias(multiaddr, &config.identity, self.context, None)
            .await?
            .value()
            .cloned()
            .ok_or_eyre("unable to resolve")?;
```